src/cli/onefuzz/backend.py

```diff
@@ -322,6 +345,13 @@ def request(
             raise Exception("endpoint not configured")
 
         url = endpoint + "/api/" + path
+
+        if (
```

Comment:
@chkeita @Porges

Reply:
I think checking client_id and authority is enough, tenant_domain isn't always required?

Reply:
Actually it is now. `tenant_domain` will always be set and used.

Reply:
if all of these value are required i would use `or` instead of `and` here

Reply:
It actually needs to be the following: 
`if self.config.client_id == "" or (
            self.config.authority == "" and self.config.tenant_domain == ""
        ):`
 
 This allows service principals to be used when the user specifies a `client_id`.